stock_landed_cost_company_percentage/models/stock_picking.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (C) 2018 - TODAY, Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    def _prepare_landed_cost_lines(self, company, amount):
+        cost_lines = []
+        for line in company.landed_cost_company_line:
+            product = line.product_id
+            account = (
+                product.property_account_expense_id.id or
+                product.categ_id.property_account_expense_categ_id.id
+            )
+            values = {
+                'product_id': product.id,
+                'split_method': product.split_method,
+                'account_id': account,
+                'price_unit': float((amount or 0.0) * line.percentage) / 100.0,
+                'name': product.name,
+            }
+            cost_lines.append((0, 0, values))
+        return cost_lines
+
+    def _prepare_landed_cost(self, amount):
+        return {
+            'picking_ids': [(6, 0, [self.id])],
+            'account_journal_id': self.company_id.journal_id.id,
```

Comment:
you should raise a UserError if the journal is not set, explaining how to fix the problem (otherwise, you get a crash because landed cost journal is required on that model)